exercises/react/src/react.h

```diff
@@ -0,0 +1,29 @@
+#ifndef REACT_H
+#define REACT_H
+
+struct reactor;
+struct cell;
+
+typedef int (*compute1) (int);
+typedef int (*compute2) (int, int);
+
+struct reactor *create_reactor();
+// free_reactor should free all cells created under that reactor.
+void free_reactor(struct reactor *);
+
+struct cell *create_input_cell(struct reactor *, int initial_value);
+struct cell *create_compute1_cell(struct reactor *, struct cell *, compute1);
+struct cell *create_compute2_cell(struct reactor *, struct cell *,
+                                  struct cell *, compute2);
+
+int cell_value(struct cell *);
+void cell_value_set(struct cell *, int new_value);
+
+typedef void (*callback) (void *, int);
+typedef int callback_id;
```

Comment:
I suppose I could say `struct callback_id` and let the student define it, but since it'd then be an incomplete type `add_callback` would have to return `struct callback *` and I didn't feel like doing more memory management for that.